### PR TITLE
cmd, consensus/ethash, eth: miner push notifications

### PR DESCRIPTION
**Pull request by karalabe**
This PR introduces a new flag `--miner.notify`, which accepts a comma-separated list of URLs to push miner work packages to. It's main use case is for mining pools to be pinged by new blocks as fast as possible, without requiring to constantly poll the `miner.getWork` endpoint.

Supersedes https://github.com/ethereum/go-ethereum/pull/17324.

### CODE REVIEW

**Comment by karalabe** [comment]
@ppratscher @rjl493456442 PTAL

File: consensus/ethash/sealer.go [hunk 1]
```diff
@@ -161,40 +164,65 @@ search:
...
+	// notifyWork notifies all the specified mining endpoints of the availability of
+	// new work to be processed.
+	notifyWork := func() {
+		work := currentWork
+		blob, _ := json.Marshal(work)
 
-	// getWork returns a work package for external miner.
+		for i, url := range notify {
+			// Terminate any previously pending request and create the new work
+			if notifyReqs[i] != nil {
```
Comment (rjl493456442) [inline consensus/ethash/sealer.go]: Should we set the `notifyReq[i]` as nil if the request has been sent successfully?

Comment (karalabe) [reply consensus/ethash/sealer.go]: That would be racey. To be fair however, this current code might be racey too. I'll think about it.

**Comment by ppratscher** [comment]
LGTM!

**Comment by rjl493456442** [comment]
Just one comment, otherwise LGTM

**Comment by karalabe** [comment]
@rjl493456442 I fixed the data race by passing `notifyReqs[i]` as a parameter to the closure instead of only the index. This way the `remote` goroutine retains ownership over `notifyReqs`. I've also added a tests that pushes a lot of work packets fast. Previously the race detector was whining, now it's happy. PTAL.

**Pull request merged**